maggy/maggy/api/routes_keys.py

```
@@ -0,0 +1,52 @@
+
+
+@router.post("")
+async def set_key(
+    request: Request, body: KeyUpdate, x_api_key: str | None = Header(None),
+) -> dict:
+    """Store or overwrite a provider key."""
+    check_auth(request, x_api_key)
+    try:
+        secrets_store.set_key(body.name, body.value, path=ENV_PATH)
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)